### DISCUSSION TITLE
Cannot upgrade: get stuck on "idealTree buildDeps"

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#14328] [Archive]
When trying to upgrade my initial Directus installation on this VPS which is 9.13.0,
I run `npm install directus --save` within the project directory.
But I get stuck on `idealTree buildDeps`

A `npm cache clear --force` did not help
The npm log says
```
38 verbose stack FetchError: request to http://registry.npmjs.org/directus failed, reason: connect ETIMEDOUT 104.16.22.35:80
38 verbose stack     at ClientRequest.<anonymous> (/usr/share/nodejs/npm/node_modules/minipass-fetch/lib/index.js:97:14)
38 verbose stack     at ClientRequest.emit (events.js:314:20)
38 verbose stack     at Socket.socketErrorListener (_http_client.js:427:9)
38 verbose stack     at Socket.emit (events.js:326:22)
38 verbose stack     at emitErrorNT (internal/streams/destroy.js:92:8)
38 verbose stack     at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
38 verbose stack     at processTicksAndRejections (internal/process/task_queues>

```
Am I doing something wrong?

npm -v	7.5.2
node -v	v12.22.12

**Comment by rijkvanzanten** [comment]
Looks like your VPS is having issues reaching npm over the internet `reason: connect ETIMEDOUT 104.16.22.35:80`. I don't think this has anything to do with Directus itself

**Reply by Pachat** [reply]
curl gives
>curl -I http://registry.npmjs.org/directus
HTTP/1.1 301 Moved Permanently
Date: Fri, 08 Jul 2022 09:18:17 GMT
Content-Type: text/plain;charset=UTF-8
Connection: keep-alive
Location: https://registry.npmjs.org/directus
Server: cloudflare
CF-RAY: 7277bb4dba0499e1-CDG

but still stuck

**Reply by azrikahar** [reply]
Perhaps you can try `npm install` another package just as a sanity check? I do agree that this seems more of a networking issue rather than Directus, at least with the current info at hand.

**Comment by Pachat** [comment]
After restarting the server and the router and an external (*) firewall it's OK now: glad to have 9.14.1 running :)

(*) Before and around the firewall on my VPS, my provider has a firewall that works like a rampart to stop unwanted ports before they arrive to the VPS.
_I deactivated it and reactivated: that is what gave the best results._
All in one, I think what I did cleared caches, which make it work
